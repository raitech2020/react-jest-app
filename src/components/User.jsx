const User = (props) => {
    const {id, name, age} = props.user
    const style = age >= 18 ? {color: 'green'} : {color: 'red'}
    console.log(id, name, age)

    return (
        <div data-testid={`user-${id}`} style={style}>{name}</div>
    )
}

export default User
