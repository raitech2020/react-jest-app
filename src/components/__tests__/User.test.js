import {render, screen, cleanup} from "@testing-library/react"
import renderer from "react-test-renderer"
import User from "../User"

test("render illegal user component", () => {
    const user = {id: 1, name: "John", age: 15}
    render(<User user={user}/>)
    // define userElement
    const userElement = screen.getByTestId(`user-${user.id}`)
    // expect userElement to be in DOM
    expect(userElement).toBeInTheDocument()
    // expect userElement to have text James
    expect(userElement).toHaveTextContent("John")
    expect(userElement).toContainHTML("red")
})

afterEach(() => {
    cleanup()
})

test("render legal user component", () => {
    const user = {id: 2, name: "Alan", age: 40}
    render(<User user={user}/>)
    // define userElement
    const userElement = screen.getByTestId(`user-${user.id}`)
    // expect userElement to be in DOM
    expect(userElement).toBeInTheDocument()
    // expect userElement to have text James
    expect(userElement).toHaveTextContent("Alan")
    expect(userElement).toContainHTML("green")
})

test("matches snapshot", () => {
    const user = {id: 1, name: "John", age: 15}
    const tree = renderer.create(<User user={user}/>).toJSON()
    expect(tree).toMatchSnapshot()
})
