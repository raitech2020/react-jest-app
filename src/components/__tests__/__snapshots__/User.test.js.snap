// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="user-1"
  style={
    Object {
      "color": "red",
    }
  }
>
  John
</div>
`;
